ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -52,55 +45,80 @@ class MoveMessage(Message):
     axis: OT3Axis
     direction: Direction
 
-    @classmethod
-    def to_message(cls, string: str) -> Optional[MoveMessage]:
-        """Parse string to Message or None."""
-        split_string = string.split(" ")
-
-        if len(split_string) == 2:
-            try:
-                axis = OT3Axis[split_string[1]]
-            except KeyError:
-                axis = None
-
-            try:
-                direction = Direction.from_symbol(split_string[0])
-            except ValueError:
-                direction = None
-
-            if axis is not None and direction is not None:
-                return cls(axis=axis, direction=direction)
-
-        return None
+    def __eq__(self, other: Any) -> bool:
+        """Confirm that the 2 objects are equal"""
+        return isinstance(other, MoveMessage) and (
+            self.axis == other.axis and self.direction == other.direction
+        )
+
+    def __hash__(self) -> int:
+        """Return hash for instance."""
+        return hash(str(self))
+
+
+class MessageLookup:
+    """Lookup class for mapping a message to a byte value."""
+
+    MESSAGE_LOOKUP_DICT: Dict[bytes, Message] = {
```

Comment:
so this isn't quite what we'd be looking for. The key here is `b"0x0000"`, which is the sequence of bytes representing the UTF-8 codepoint sequence "0x0000", which is the bytes (note, no quotes) 0x307830303030 , 6 bytes total - one for each character.

What we want is the hex sequence 0x0000.

This can be represented in python with the sequence `b"\x00\x00"` using hex escapes (see [https://docs.python.org/3/reference/lexical_analysis.html](the docs) in the "Escape Sequences" table which for some reason doesn't have a direct link anchor), and 0x0001 could be `b"\x00\x01"`, etc; but this gets a little clunky, and instead i'd recommend using the `[struct](https://docs.python.org/3/library/struct.html#struct.pack)` module, whose job is to pack data into and parse data out of byte strings.

Typically you wouldn't want to put byte strings in lookups like this directly; rather, you'd take the incoming string and do `struct.unpack('<H', inputbytes)`, which would tell struct to interpret the inputbytes as a two-byte short integer, and turn it into a python integer, which could then be a struct key.